src/Gamecube.c

```diff
@@ -99,23 +138,31 @@ uint8_t gc_write(const uint8_t pin, Gamecube_Status_t* status, Gamecube_Origin_t
     uint8_t command[3];
     uint8_t receivedBytes = gc_n64_get(command, sizeof(command), modePort, outPort, inPort, bitMask);
 
-    // Init
-    if (receivedBytes == 1 && command[0] == 0x00)
+    // Init or reset
+    if (receivedBytes == 1 && (command[0] == 0x00 || command[0] == 0xFF))
     {
         gc_n64_send(status->raw8, sizeof(Gamecube_Status_t), modePort, outPort, bitMask);
         ret = 1;
     }
-    // Get origin
-    else if (receivedBytes == 1 && command[0] == 0x41)
+    // Get origin or recalibrate
+    else if (receivedBytes == 1 && (command[0] == 0x41 || command[0] == 0x42))
     {
         gc_n64_send(origin->raw8, sizeof(Gamecube_Origin_t), modePort, outPort, bitMask);
         ret = 2;
     }
     // Get data. Do not check last byte (command[2]), as the flags are unknown
-    else if (receivedBytes == 3 && command[0] == 0x40 && command[1] == 0x03)
+    else if (receivedBytes == 3 && command[0] == 0x40 && command[1] <= 0x07)
     {
-        gc_n64_send(report->raw8, sizeof(Gamecube_Report_t), modePort, outPort, bitMask);
-        ret = 3;
+        if (command[1] == 0x03)
+        {
+            gc_n64_send(report->raw8, sizeof(Gamecube_Report_t), modePort, outPort, bitMask);
+            ret = 3;
+        } else {
+            Gamecube_Different_Report_t dif;
+            gc_report_convert(report, &dif, command[1]);
+            gc_n64_send(dif.raw8, sizeof(Gamecube_Different_Report_t), modePort, outPort, bitMask);
+            ret = 3;
+        }
```

Comment:
Now the question is, if we can simplify this by adding a return on mode 3 into the convert function and always send the `dif`. Like:

```c
Gamecube_Different_Report_t dif;
gc_report_convert(report, &dif, command[1]);
gc_n64_send(dif.raw8, sizeof(Gamecube_Different_Report_t), modePort, outPort, bitMask);
ret = 3;
```

And this in the convert function:
```c
    else if (mode == 3)
    {
        return;
    }
```

The downside is, that for normal reading mode an additional memcpy call is made. I am not sure if that additional time will cause issues, but I guess it will work, if it also works for the other modes that do additional move operations. Could you please compare the flash size?

Reply:
Did you check the flash size difference? Was it a good change?

Reply:
I'm not very good at English, and I'm not sure I fully read your intent... (This text is also written in translation.)
What do you mean by flash?
Which structure are you referring to? (mode0~4? Gamecube_Different_Report_t?)

Reply:
Oh sorry, I was not aware of that.

By flash I mean the flash memory. You can check it like that:
![image](https://user-images.githubusercontent.com/6888294/125067625-cd6add80-e0b4-11eb-929f-d67a03ee6a4f.png)

My question is, if the flash memory size increased? Was it a good or bad change?

Reply:
I understood.
"mode3=return" in the conversion function seems to reduce the size of the flash on the contrary.
I tried it with mode0 going through the function, but it didn't cause any noticeable delay.